### PR TITLE
Bumped OpenTelemetry Collector to v0.14.0

### LINKED ISSUE #111: Update to collector v0.14.0
The collector v0.14.0 was being released yesterday. This issue is to track the update to this version, including the upgrade procedure based on the changelog.

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #111

﻿Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
@dengliming, would you like to review this one?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Changes by Version
 ==================
 
+0.14.0 (2020-11-09)
+-------------------
+* Bumped OpenTelemetry Collector to v0.13.0 ([#101](https://github.com/open-telemetry/opentelemetry-operator/pull/101), [@jpkrohling](https://github.com/jpkrohling))
```
Comment (dengliming) [inline CHANGELOG.md:6]: Why is this the same as the `v0.13.0` log?  should be `Bumped OpenTelemetry Collector to v0.14.0 ([#112](https://github.com/open-telemetry/opentelemetry-operator/pull/112)`  Is that right?

Comment (jpkrohling) [reply CHANGELOG.md]: I typically first open the PR and then amend with the right links, but forgot this last step this time.

**Review by dengliming — APPROVED** [review]
👍 Looks great

**Pull request merged**